### ISSUE TITLE
Hi Please tell me about the description of the dataset of Product.stats.csv of eShopDashboard Machine Learning Project of Sales Forecasting  like

### ISSUE THREAD

**Issue by Aamirkhan10218** [dotnet/machinelearning#5126]
Hi Please tell me about the description of the dataset of Product.stats.csv of eShopDashboard Machine Learning Project of Sales Forecasting  like 
what is the meaning of 
next
avg
count
max
min
prev
[products.stats.xlsx](https://github.com/dotnet/machinelearning-samples/files/4628273/products.stats.xlsx)
i'll be very grateful to you

_Originally posted by @Aamirkhan10218 in https://github.com/dotnet/machinelearning-samples/pull/661#issuecomment-628601541_

**Comment by antoniovs1029** [comment]
Hi, you can find explanations of the dataset in here:
[Details of the Dataset
](https://github.com/dotnet/machinelearning-samples/blob/master/samples/csharp/end-to-end-apps/Forecasting-Sales/docs/Details-of-Dataset.md)

[Description of the sample (and how it uses the dataset)
](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/Forecasting-Sales)

Also, notice that the dataset comes from this source, where you might be able to find more information:
http://archive.ics.uci.edu/ml/datasets/online+retail#

**Comment by Aamirkhan10218** [comment]
Thank you for your response. I am very grateful to you.

**Comment by Aamirkhan10218** [comment]
> i have one more confusion. kindly tell me that on which basis Online
>> Retail Data set transform into Product.stats.csv file on attributes
>
> like next,prev,min,max.
i mean how you calculate next parameter from Online Retail Dataset.
i am using this technique of sales forecasting in my Final Year Project
kindly clear my confusion or share any link which explain eShopDashboard
Project.
Thank you so much

**Comment by antoniovs1029** [comment]
Hi @Aamirkhan10218 . Sorry, I am not sure what you're asking for.

Do you mean to ask how does the eShopDashboard app works?
In which case, the best I can provide is the description I've already linked to, as well as recommending looking into the code:
https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/Forecasting-Sales

Or, do you mean to ask how does the sales forecasting algorithm works? Or perhaps something else?

**Comment by Aamirkhan10218** [comment]
yeah please tell me how Algorithm work  also tell me that how dataset is is
used with algorithm. let me clear there is a **next** attribute in the
dataset just tell me that how  it come from the base dataset(Online Retail).

**Comment by antoniovs1029** [comment]
Hi, @Aamirkhan10218 .

It seems that we don't provide any docs explaining the algorithm (except for the high-level explanations of the estimaror's parameters). So I can only recommend looking for external sources such as the wikipedia article [on the algorithm](https://en.wikipedia.org/wiki/Singular_spectrum_analysis).

Regarding your question about the dataset, if you look into the `next` and `prev` columns you'll noticed that they are the same, only that they're shifted by 3 steps. So what I think that happened is that simply they counted the number of items sold on a given month, and then put that number on the `next `and `prev `column to indicate how many items were sold in the next and previous month. So, no special algorithm for this... it was simply done to generate a dataset that would have the necessary columns to train the SSA forecaster, and you wouldn't need to worry about it if you already have a dataset with such columns.

It's odd, though, that the `products.stats.xlsx` file mentions months between 2015-2017, whereas the original `Online Retail.xlsx` dataset only mentions 2010-2011. So I've asked the docs team to take a look into this.

**Comment by antoniovs1029** [comment]
Since this is actually a question specific to the samples, I'll move this to the samples repo:
https://github.com/dotnet/machinelearning-samples
